source/common/runtime/runtime_protos.h

```diff
@@ -11,6 +11,27 @@
 namespace Envoy {
 namespace Runtime {
 
+// Helper class for runtime-derived uint32.
+class UInt32 {
+public:
+  UInt32(const envoy::config::core::v3::RuntimeUInt32& uint32_proto, Runtime::Loader& runtime)
+      : runtime_key_(uint32_proto.runtime_key()), default_value_(uint32_proto.default_value()),
+        runtime_(runtime) {}
+
+  const std::string& runtimeKey() const { return runtime_key_; }
+
+  uint32_t value() const {
+    uint64_t raw_value = runtime_.snapshot().getInteger(runtime_key_, default_value_);
+    return raw_value > std::numeric_limits<uint32_t>::max() ? default_value_
```

Comment:
This may introduce additional complexity. If we introduce additional variables to record the number of invalid parsing, this class will become thread-unsafe, making it either complicated or poor in performance. 
I'm not sure if there are other solutions, I look forward to your suggestions.

Reply:
I mean just put ENVOY_LOG_EVERY_POW_2()  or ENVOY_LOG_EVERY_N() in a conditional instead of using that ternary operator. You don't need to get too clever here. 

I would be genuinely surprised if that caused an appreciable decline in performance.

Reply:
I misunderstood what you meant before (embarrassing >_<). Now I understand your suggestion and submit a new commit.